### ISSUE TITLE
mysqli class doesn't always report SQL errors properly or at all

### ISSUE THREAD

**Issue by eSilverStrike** [Geeklog-Core/geeklog#764]
So I have upgraded a server from PHP 5.6 to PHP 7.1

In the error log I notice the following error every so often:
```
Call to a member function fetch_array() on boolean @ ...\system\databases\mysqli.class.php line 829
```
Not much to go on but I eventually tracked down the cause of the SQL error to the gus plugin. The GUS plugin executes code whenever the plugins functions.inc is included (when included by the plugins library by lib-common). There is nothing really special about the code which errors out on line 164 (the DB_fetchARRAY line)
```
$rec = DB_query("SELECT COUNT(*) AS ignored
	FROM {$_TABLES['gus_ignore_referrer']}
	WHERE '{$referrer}' LIKE referrer
	LIMIT 1", 1);
$row = DB_fetchArray($rec, FALSE);
```
The sql statement was failing when the referrer was something which caused the sql statement to fail but that is not important for this issue report.

**What is important** is that when this sql statement fails the above error gets written into the error log and not the standard error like below that shows the sql statement causing the problem:
```
You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ...
```
I also notice that with PHP 7.1 when the sql statement failed (I made it fail on purpose by changing the select statement) the user is presented with the correct type of error page:

Unfortunately, an error has occurred rendering this page. Please try again later. 

but when I tried the same process under PHP 5.6 by making the same SQL statement failed the user is presented with a blank page and nothing is written into the Geeklog error log.

I am not sure why the error is not being reported properly  in PHP 7.1 or at all in PHP 5.6 but there may be some sort of explanation here: http://stackoverflow.com/questions/37455186/fatal-error-call-to-a-member-function-fetch-array-on-boolean

I also wonder if it has something to do with the fact the GUS functions.inc is included early in the page execution, as when I introduce other sql errors in for example Geeklogs index.php, it is reported in the error log properly.

**Comment by eSilverStrike** [comment]
Also seems related to this issue: https://github.com/Geeklog-Plugins/forum/issues/28

**Comment by mystralkk** [comment]
I think I have fixed this bug with change set fe4a33b.  Can you check if it is fixed?

**Comment by eSilverStrike** [comment]
Jus a note regarding the above I also see that in the above code from the Gus plugin, the function DB_query is passed with a 1 which ignores all errors if they happen. Which makes the error I had experienced before a little more weird since it should not have been reported at all.

Anyways I did run some limited tests and things look like they are working but I can't 100% confirm until I test on a live site with a lot of traffic (which the development 2.2.0 version of Geeklog is not ready for yet). Should we close this issue for now or leave it open until the beta version of Geeklog 2.2.0 is ready for testing on a live site?
